### ISSUE TITLE
sensu-api service should subscribe to sensu::rabbitmq::config class

### ISSUE THREAD

**Issue by fatmcgav** [sensu/sensu-puppet#433]
It appears that currently the `sensu-api` service [1] doesn't subscribe to the `sensu::rabbitmq::config` class, which means that it doesn't get restarted automatically if the rabbitmq config is changed. 

From my testing I believe it should do, otherwise the API doesn't work. 

[1] https://github.com/sensu/sensu-puppet/blob/master/manifests/api/service.pp#L34-L39

**Comment by jaxxstorm** [comment]
The API doesn't actually need any interaction with rabbitmq, so I don't think this is needed.

Closing, reopen if you feel this is in error.

**Comment by stevenviola** [comment]
I looks like the API does do something with rabbitmq (in Sensu 0.26.5), and not refreshing the sensu-api after updating `rabbitmq.json` can leave sensu-api in a broken state. To test this, I modified `rabbitmq.json` to connect to the wrong port and restarted the sensu-api. In the sensu-api logs were these messages:

```
{"timestamp":"2016-12-30T18:32:20.081877+0000","level":"error","message":"[amqp] Detected TCP connection failure: Errno::ECONNREFUSED"}
{"timestamp":"2016-12-30T18:32:23.848442+0000","level":"info","message":"api response","request":{"remote_address":"172.31.13.168","user_agent":"Go-http-client/1.1","method":"GET","uri":"/stashes","query_string":null,"body":""},"status":500,"content_length":59}
```

Running puppet put the right information back into `rabbitmq.json`, and triggered the sensu-server and sensu-client to be restarted, but not the sensu-api. This left sensu-api in a broken, returning 500 status for every request, until it was manually restarted to pick up the rabbitmq changes.

I'm not sure why sensu-api needs to connect to rabbitmq, but for whatever reason, it looks like sensu-api should be restarted whenever rabbitmq.json is updated to pull in the new values. Adding the api service to subscribe to `sensu::rabbitmq::config`, as @fatmcgav suggests, fixes the issue for me

**Comment by cwjohnston** [comment]
@jaxxstorm the API [establishes a connection to the transport](https://github.com/sensu/sensu/blob/v0.26.5/lib/sensu/api/process.rb#L19). Uses for this connection include: 

* reporting number of consumers for `/info` and `/health` endpoints
* publishing ad-hoc check requests via `/request` endpoint

**Comment by ghoneycutt** [comment]
@jeffmccune This also looks very similar to ae4885952830868e346eec60b52acfa4c8a28853

**Comment by ghoneycutt** [comment]
Thanks everyone!

Fix released in v2.19.2
